### DISCUSSION TITLE
advanced search & search in transactions

### DISCUSSION THREAD

**Discussion by TWART016** [firefly-iii/firefly-iii#5140] [General]
Hi,

I know that on the top left is a search bar. But filter for specific source accounts (source_account_is:query) or has a tag (tag_is:query) is very difficult, special for non technical users or on mobile phones.

Is it possible to create an advanced search page?

Also it would be great if there is a search on the transaction page to easily find something.

**Comment by JC5** [comment]
I've put them on the list :+1:

**Reply by TWART016** [reply]
Thanks you.

Sort and Filter inside the "tables" (e.g. in transactions) would also be great.

**Comment by TWART016** [comment]
That could look something like that
![image](https://user-images.githubusercontent.com/29805669/149297477-b26f2018-07f7-4184-bc30-1e3da30e2179.png)

Timepicker could be that
![image](https://user-images.githubusercontent.com/29805669/149296485-230eac36-35e3-45f2-8402-af56afab6b9e.png)

In the top bar are the most used filters.

The Operator could be variable dependent on the Filter, e.g. Filter Amount = Equals, greater, smaller or Tag = Is empty, Is Not empty, contains, start/ends with, (Does not) contains. 
Same for Value column.

The Tools on the top right could contain Save Search (for quick access), select colums for search results, copy/paste, export data, subscripe (automatic e-mail report).

**Reply by JC5** [reply]
Honestly, that's the kind of user interface I'm trying to avoid. 

![simplicity](https://user-images.githubusercontent.com/5889984/149298947-981c0488-bf36-4e37-bf70-1256e8c2d4ab.png)

**Comment by TWART016** [comment]
Of course it can also look like google in Simple mode
![image](https://user-images.githubusercontent.com/29805669/149299614-5ee011a9-f078-4a82-a3ba-27bb7a92de0b.png)

But thinking of the right command is quite difficult. It does not have to be as detailed as in the example.

The apple model `Touch` could be done with `Save Search`
